src/components/views/settings/Notifications.tsx

```diff
@@ -538,8 +494,12 @@ export default class Notifications extends React.PureComponent<IProps, IState> {
             } else {
                 const definition: VectorPushRuleDefinition = VectorPushRulesDefinitions[rule.ruleId];
                 const actions = definition.vectorStateToActions[checkedState];
-                await this.setPushRuleActions(rule.rule.rule_id, rule.rule.kind, actions);
-                await this.updateSyncedRules(definition.syncedRuleIds, actions);
+                // we should not encounter this
+                if (!rule.rule) {
+                    throw new Error("Unable to update rule without rule_id");
```

Comment:
The message is mentions `rule_id` while the condition `!rule.rule` does not validate it.
Can you update either the message or the condition, so that they match?